art/attacks/evasion/pixel_threshold.py

```diff
@@ -152,9 +144,9 @@ def generate(self, x, y=None, maxiter=100, **kwargs):
                 self.min_th = 127
                 start, end = 1, 127
                 while True:
-                    image_result = []
+                    image_result: Union[List[np.ndarray], np.ndarray] = []
                     threshold = (start + end) // 2
-                    success, trial_image_result = self._attack(image, target_class, threshold, maxiter)
```

Comment:
Can `maxiter` really be removed?

Reply:
[TODO] add it back and pass it on to optimizer in `_attack`.

Reply:
Added back as `max_iter`, passed to the solver.